features/netsocket/emac-drivers/TARGET_ARM_SSG/COMPONENT_SMSC9220/smsc9220_emac.cpp

```diff
@@ -0,0 +1,376 @@
+/*
+ * Copyright (c) 2018 Arm Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <ctype.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "cmsis_os.h"
+
+#include "mbed_interface.h"
+#include "mbed_wait_api.h"
+#include "mbed_assert.h"
+#include "netsocket/nsapi_types.h"
+#include "mbed_shared_queues.h"
+
+#include "smsc9220_emac.h"
+#include "smsc9220_eth_drv.h"
+
+/** \brief Defines for receiver thread */
+#define FLAG_RX                              1U
+#define LINK_STATUS_THREAD_PRIORITY          (osPriorityNormal)
+#define LINK_STATUS_THREAD_STACKSIZE         2048U
+#define LINK_STATUS_TASK_PERIOD_MS           200U
+#define PHY_STATE_LINK_DOWN                  0U
+#define PHY_STATE_LINK_UP                    1U
+#define PHY_REG_BSTATUS_LINK_STATUS_INDEX    2U
+#define CRC_LENGTH_BYTES                     4U
+
+#ifndef SMSC9220_ETH
+#error "SMSC9220_ETH should be defined, check device_cfg.h!"
+#endif
+static const struct smsc9220_eth_dev_t *dev = &SMSC9220_ETH_DEV;
+/* Processing thread */
+static osThreadId_t receiver_thread = 0;
+
+SMSC9220_EMAC::SMSC9220_EMAC()
+{
+}
+
+static osThreadId_t create_new_thread(const char *threadName,
+                                      void (*thread)(void *arg), void *arg,
+                                      int stacksize, osPriority_t priority,
+                                      mbed_rtos_storage_thread_t *thread_cb)
+{
+    osThreadAttr_t attr = {0};
```

Comment:
We prefer use of the C++ wrappers to direct CMSIS-RTOS use, so `Thread` class and `ThisThread` namespace rather than `osThreadxxx` calls.